packages/x-data-grid/src/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -183,9 +183,12 @@ export const useGridPrintExport = (
       const gridFooterElement: HTMLElement | null = gridClone.querySelector(
         `.${gridClasses.footerContainer}`,
       );
-      gridFooterElement!.style.position = 'absolute';
-      gridFooterElement!.style.width = '100%';
-      gridFooterElement!.style.top = `${computedTotalHeight - gridFooterElementHeight}px`;
+
+      if (gridFooterElement) {
+        gridFooterElement.style.position = 'absolute';
+        gridFooterElement.style.width = '100%';
+        gridFooterElement.style.top = `${computedTotalHeight - gridFooterElementHeight}px`;
+      }
```

Comment:
If it's related to `options.hideFooter` only, would it be better instead to move this code to the else block of `if (normalizeOptions.hideFooter) {` (line 165) to avoid the need to find this element at all when `hideFooter` is `true`?

Reply:
@MBilalShafi, `gridFooterElement.style.top = ${computedTotalHeight - gridFooterElementHeight}px; ` depends on `computedTotalHeight` which is calculated after  the `if (normalizeOptions.hideFooter) { (line 165)` condition. 
Should I just wrap this code in a `if (!normalizeOptions.hideFooter) {` ?

Reply:
Sorry about the late response. Yes I think that makes sense. I committed the code to speed things up.
This removes the need to find the DOM element `gridFooterElement` if the `hideFooter` option is set.